zephyr/Kconfig

```diff
@@ -56,4 +56,13 @@ config ZEPHYR_DP_SCHEDULER
 	  DP modules can be located in dieffrent cores than LL pipeline modules, may have
 	  different tick (i.e. 300ms for speech reccognition, etc.)
 
+config INLINE_FUNCTIONS
+	bool
+	default y if CAVS_VERSION_2_5 || ACE
+	default n
+	depends on CAVS_VERSION_2_5
+	depends on ACE
```

Comment:
I don't think you need these two dependencies

Reply:
I added based on previous config, like ZEPHYR_DP_SCHEDULER, so not 100% sure, maybe you are right.

Reply:
`ZEPHYR_DP_SCHEDULER` is user-selectable - it has a comment after the type field, and its "default y if X" and "depends on Y" are different:
```
config ZEPHYR_DP_SCHEDULER
	bool "use Zephyr thread based DP scheduler"
	default y if ACE
	default n
	depends on IPC_MAJOR_4
	depends on ZEPHYR_SOF_MODULE
```
Whereas here it's strange that the same config options are used in "default y if ..." and in "depends on ..." statements. In principle it makes sense: (1) it sets a default, and (2) it prevents overrides to "y" by other platforms, but is that needed? Also, don't those two dependencies mean, that it depends on both enabled at the same time, which is never the case? I would just remove those dependencies - if somebody wants it, I don't see why they shouldn't be able to try it.